recipes/ecfeed/all/test_package/conanfile.py

```diff
@@ -0,0 +1,16 @@
+import os
+from conans import ConanFile, CMake, tools
+
+class HelloTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)       
+        cmake.configure()
+        cmake.build()          
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            os.chdir("bin")
+            self.run(".%sexample" % os.sep)
```

Comment:
```suggestion
        if not tools.cross_building(self):
            bin_path = os.path.join("bin", "example")
            self.run(bin_path, run_environment=True)
```

self.settings is no longer used to crossbuilding

Reply:
Updated.